### PR TITLE
[TRAFODION-2873] Removed usage of system new from LobGlobals and other classes

### PR DESCRIPTION
**Pull request by sandhyasun**
ExpOBInterfaceInit will take a parent heap and derive a lobHeap from it. ExLobGlobals will get created from this heap and store this heap pointer for use by other classes.
ExpLOBinterfaceCleanup  destroys the LOB globals and will utilize the heap pointer stored within it. 
In cli/Cli.cpp , Removed the initialization/destruction  of LOB glabals from the ::createLOB interface and moved it out before the interface is called to be consistent with other LOB API usage.

### CODE REVIEW

File: core/sql/exp/ExpLOBaccess.h [hunk 1]
```diff
@@ -589,7 +589,7 @@ class ExLobGlobals
 {
   public :
   
-    ExLobGlobals(); 
+    ExLobGlobals(NAHeap *lobHeap=NULL);
```
[the remark this answers is not in the document]
Comment (DaveBirdsall) [reply core/sql/exp/ExpLOBaccess.h]: To @selvaganesang's point, perhaps we should leave this without a default, and then code NULL explicitly in ExpLobProcess.cpp?

File: core/sql/cli/Cli.cpp [hunk 2]
```diff
@@ -9611,12 +9611,28 @@ Lng32 SQLCLI_LOBddlInterface
 	    
 	  } // for
 
+        //Initialize LOB interface 
+        
+        Int32 rc= ExpLOBoper::initLOBglobal(exLobGlob,currContext.exHeap(),&currContext,hdfsServer,hdfsPort);
+        if (rc)
+          {
+            {
```
Comment (DaveBirdsall) [inline core/sql/cli/Cli.cpp]: It doesn't hurt anything, but I'm curious: Why two sets of braces? Looks like one would do?

File: core/sql/cli/Cli.cpp [hunk 3]
```diff
@@ -9726,10 +9742,26 @@ Lng32 SQLCLI_LOBddlInterface
         //lob data files.  Note that if there is an error in the drop of the 
         //descriptor tables above , the transaction will restore each of the 
         //above tables . 
+        //Initialize LOB interface 
+       
+        Int32 rc= ExpLOBoper::initLOBglobal(exLobGlob,currContext.exHeap(),&currContext,hdfsServer,hdfsPort);
+        if (rc)
+          {
+            {
```
Comment (DaveBirdsall) [inline core/sql/cli/Cli.cpp]: Same comment here

File: core/sql/cli/Cli.cpp [hunk 4]
```diff
@@ -9765,12 +9797,27 @@ Lng32 SQLCLI_LOBddlInterface
 	    
 	    goto error_return;
 	  }
-	
+	//Initialize LOB interface 
+        
+        Int32 rc= ExpLOBoper::initLOBglobal(exLobGlob,currContext.exHeap(),&currContext,hdfsServer,hdfsPort);
+        if (rc)
+          {
+            {
```
Comment (DaveBirdsall) [inline core/sql/cli/Cli.cpp]: And here

File: core/sql/exp/ExpLOBaccess.h (hunk 1 above)
[the remark this answers is not in the document]
Comment (sandhyasun) [reply core/sql/exp/ExpLOBaccess.h]: Sure that's another way. Will do that.

File: core/sql/exp/ExpLOBprocess.cpp [hunk 5]
```diff
@@ -575,7 +575,7 @@ Lng32 main(Lng32 argc, char *argv[])
     // setup log4cxx
     QRLogger::initLog4cxx(QRLogger::QRL_LOB);
     // initialize lob globals
-    lobGlobals = new ExLobGlobals();
+    lobGlobals = new ExLobGlobals(NULL);
```
Comment (selvaganesang) [inline core/sql/exp/ExpLOBprocess.cpp]: Thanks for avoiding the default value for the param. But, I think sending heap as NULL might make this program to fail. Have you tested this program? If you have tested this program already, then this PR is ready to be merged.

Comment (sandhyasun) [reply core/sql/exp/ExpLOBprocess.cpp]: The ExpLOBprocess.cpp is dormant code for now. If we enable it and create a LOB process to offload some operations like GC etc, we will need to create and setup globals , send messages form the master process to it etc .  For now I just had to ensure it compiled.

**Pull request merged**